### PR TITLE
sync with google calendar

### PR DESCRIPTION
**Pull request by clementtalleu**
# Description

Allow a repairer to connect its google account to register rustine libre's calendar's events in its google calendar

# Changes

| Q             | A        
|---------------| ---------
| Issue         | 
| Type          | <ul><li>- [x] Feat</li><li>- [ ] Hotfix</li><li>- [ ] Doc</li><li>- [ ] Refacto</li></ul>
| Break changes |  No

### CODE REVIEW

File: pwa/pages/_app.tsx [hunk 1]
```diff
@@ -36,179 +37,184 @@ function MyApp({
...
-                <A2HSIOS></A2HSIOS>
-                <Notifications></Notifications>
-                <Layout dehydratedState={pageProps.dehydratedState}>
-                  <Component {...pageProps} />
-                </Layout>
-              </CssBaseline>
-            </ThemeProvider>
-          </>
-        </UserFormProvider>
+        <GoogleOAuthProvider clientId="821363302500-6nc8cl1jseugkcl1n9lqjv64rueotfad.apps.googleusercontent.com">
```
Comment (davy-beauzil) [inline pwa/pages/_app.tsx:40]: Use env variable instead

**Review by davy-beauzil — COMMENTED** [review]
I don't know if this is due to your work, but sometimes, there are some appointments are not showed in the agenda.
- I have an appointment who  is appearing in the response of the request, but who is not showed in the agenda
- I have an appointment who is not appearing in the response of the request, when it should be present

It will be fine to export an appointment when it is created, not only when we login with our google account.

When I try to logout (I delete the cache) and re log in with my google account, an appointment is marked "sync" in the database, but it not appears in my google agenda.

**Review by davy-beauzil — APPROVED** [review]

**Comment by clementtalleu** [comment]
> It will be fine to export an appointment when it is created, not only when we login with our google account.

@davy-beauzil it s the case everytime a repairer accepts an appointment

File: api/src/Repository/AppointmentRepository.php [hunk 2]
```diff
@@ -117,4 +117,19 @@ public function getThreeLastAppointmentsRepairersIds(User $user): array
 
         return array_column($qb->getQuery()->getArrayResult(), 'repairerId');
     }
+
+    public function getNextAppointmentsNotSync(Repairer $repairer): array
+    {
+        $qb = $this->createQueryBuilder('a')
+            ->leftJoin('a.repairer', 'r')
```
[the remark this answers is not in the document]
Comment (clementtalleu) [reply api/src/Repository/AppointmentRepository.php:124]: no it's not I remove it

**Pull request merged**